Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -8,6 +8,460 @@ namespace gdjs {
    * considered as a platform by objects having PlatformerObject Behavior.
    */
   export class PlatformerObjectRuntimeBehavior extends gdjs.RuntimeBehavior {
+    /**
+     * The object is on the floor standing or walking.
+     */
+    OnFloor = class implements gdjs.PlatformerObjectRuntimeBehavior.State {
```

Comment:
I missed that you changed this to put "class expressions" inside the behavior class.
I would highly recommend against this:
* This makes the class appear bloated.
* This force the JS engine to make every single instance of the class to have its own class.

The second part can be seen like this:
![image](https://user-images.githubusercontent.com/1280130/111442258-10690580-8700-11eb-8c86-c8e394211504.png)

Here `A.test` is not accessible because `test` is a member of the class, so you have to instantiate a instance of A... and each instance of A will get its own class!
This means if you have 1500 objects using the platform behavior, there will be 1500 OnFloor classes in memory.

Instead, you can declare classes but not export them:

```
namespace gdjs {
  class OnFloor implements .... { ... }
  class Falling implements .... { ... }

  export class PlatformerObjectRuntimeBehavior { ... }
}
```

In other words, you can declare any class or variable inside namespace gdjs { } and it will be scoped to the file - which is great, nobody will be ever able to access it. Then when you use the keyword `export` on a variable or a class, it will become accessible under `gdjs.YourClass` (`or gdjs.yourVariable`).

Reply:
> (note that you could make these `static`, but at this point it's not better than declaring the state classes outside of the behavior class, as it still make the behavior class appear bloated :))

It seems a good option because it allows to keep private what should be. With the class without export, there is a lot of members that can't be private but should.
I have both version in local, what do you prefer?